lc-hax/Scripts/Patches/GiftBoxPatch.cs

```diff
@@ -4,8 +4,8 @@
 
 [HarmonyPatch(typeof(GiftBoxItem), nameof(GiftBoxItem.ItemActivate))]
 class GiftBoxPatch {
-    static void Prefix(ref bool used, ref bool ___hasUsedGift) {
-        used = false;
-        ___hasUsedGift = false;
+    static bool Prefix(ref GiftBoxItem __instance) {
+        __instance.OpenGiftBoxServerRpc();
+        return false;
```

Comment:
It doesn't make sense to me how this solves the problem but it's amazing you managed to find this out. Great job!

Reply:
Seems like this method is causing some exceptions..

![image](https://github.com/winstxnhdw/lc-hax/assets/56998716/a168ae71-8f93-4e17-af6e-368c218f468c)

Reply:
because the gift status gets changed to others and the local client will result unchanged.

Reply:
> Seems like this method is causing some exceptions..
> 
> ![image](https://private-user-images.githubusercontent.com/56998716/298203771-a168ae71-8f93-4e17-af6e-368c218f468c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3MDA4ODAsIm5iZiI6MTcwNTcwMDU4MCwicGF0aCI6Ii81Njk5ODcxNi8yOTgyMDM3NzEtYTE2OGFlNzEtOGY5My00ZTE3LWFmNmUtMzY4YzIxOGY0NjhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDIxNDMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhNjk2Yjc3MjkwNTIxZTEzNDhiM2Y4ZDQ2ZjFkZTI0YjI0ZmU4NWIyZDgwMmJiOWZiODJlZTM3ZWEwMGIyMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iC90OsBGjnnBNXIKMNDnZE4eD_9f58uLQrcoNODryfU)

yeah because the particles get nullified, it can be easily fixed by remaking that RPC and do just the spawn part.